x/subspaces/keeper/msg_server.go

```diff
@@ -150,6 +150,218 @@ func (k msgServer) DeleteSubspace(goCtx context.Context, msg *types.MsgDeleteSub
+	section, found := k.GetSection(ctx, msg.SubspaceID, msg.SectionID)
+	if !found {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "section with id %d not found inside subspace %d", msg.SectionID, msg.SubspaceID)
+	}
+
+	signer, err := sdk.AccAddressFromBech32(msg.Editor)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid editor address: %s", msg.Editor)
+	}
+
+	// Check the permission to manage sections
+	if !k.HasPermission(ctx, msg.SubspaceID, types.RootSectionID, signer.String(), types.PermissionManageSections) {
+		return nil, sdkerrors.Wrap(types.ErrPermissionDenied, "you cannot manage sections within this subspace")
+	}
+
+	// Update the section and validate it
+	update := types.NewSectionUpdate(msg.Name, msg.Description)
+	updated := section.Update(update)
+	err = updated.Validate()
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, err.Error())
+	}
+
+	// Save the section
+	k.SaveSection(ctx, updated)
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			sdk.NewAttribute(sdk.AttributeKeyAction, sdk.MsgTypeURL(msg)),
+			sdk.NewAttribute(sdk.AttributeKeySender, msg.Editor),
+		),
+		sdk.NewEvent(
+			types.EventTypeEditSection,
+			sdk.NewAttribute(types.AttributeKeySubspaceID, fmt.Sprintf("%d", section.SubspaceID)),
+			sdk.NewAttribute(types.AttributeKeySectionID, fmt.Sprintf("%d", section.ID)),
+		),
+	})
+
+	return &types.MsgEditSectionResponse{}, nil
+}
+
+// MoveSection defines a rpc method for MsgMoveSection
+func (k msgServer) MoveSection(goCtx context.Context, msg *types.MsgMoveSection) (*types.MsgMoveSectionResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check the if the subspace exists
+	if !k.HasSubspace(ctx, msg.SubspaceID) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "subspace with id %d not found", msg.SubspaceID)
+	}
+
+	// Check if the section exists
+	section, found := k.GetSection(ctx, msg.SubspaceID, msg.SectionID)
+	if !found {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "section with id %d not found inside subspace %d", msg.SectionID, msg.SubspaceID)
+	}
+
+	// Check if the destination section exists
+	if !k.HasSection(ctx, msg.SubspaceID, msg.NewParentID) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "section with id %d does not exist inside subspace %d", msg.NewParentID, msg.SubspaceID)
+	}
+
+	signer, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid signer address: %s", msg.Signer)
+	}
+
+	// Check the permission to manage sections
+	if !k.HasPermission(ctx, msg.SubspaceID, types.RootSectionID, signer.String(), types.PermissionManageSections) {
+		return nil, sdkerrors.Wrap(types.ErrPermissionDenied, "you cannot manage sections within this subspace")
+	}
+
```

Comment:
We should avoid setting parentID as itself.
Currently the following process is accepted:
1. Create a section whose ID = 1 and parentID = 0
2. Move the section to make itself as the parent section (ID = 1 and parentID = 1)

Reply:
Thanks, I didn't spot this! I've added a check both inside the `msg_server` as well as inside the `Section` type to make sure it's not possible to have a section which `ID` and `ParentID` are the same